R/commented_code_linter.R

```diff
@@ -0,0 +1,29 @@
+#' @describeIn linters checks that there is no commented code outside roxygen
+#'   blocks
+#' @export
+commented_code_linter <- function(source_file) {
+  res <- re_matches(source_file$file_lines,
+                    rex("#", except("'"), anything,
+                        or(some_of("{}[]"), # code-like parentheses
+                           "<-", "->", "==", # an assignment
```

Comment:
Should this check for `=` assignment rather than `==` equality?

I assume you restricted the search to assignments only because it could be valid to have a comment like

```r
# checks that var > 2
if (var > 2) {
  x
}
```